### PR TITLE
Adapt to hive s3 changes

### PR DESCRIPTION
**Pull request by maltesander**
## Description

Adapted tests for https://github.com/stackabletech/hive-operator/pull/179
Added test for external table (S3)

Currently requires a certain secret operator pr 
`helm install secret-operator stackable-test/secret-operator --version=0.3.0-pr125`

## Review Checklist
- [ ] Code contains useful comments
- [ ] Changelog updated (or not applicable)

### CODE REVIEW

File: hive-operator/tests/kuttl/smoke/00-assert.yaml [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
 kind: TestAssert
 metadata:
-  name: install-test-hive-derby
-timeout: 600
+  name: install-minio
+timeout: 900
 ---
-apiVersion: apps/v1
-kind: StatefulSet
+apiVersion: v1
```
Comment (sbernauer) [inline hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: what exactly are you checking here? Does it work to specify the Service and kuttl waits for the included Pods to become ready?
(same for `01-assert.yaml`)

File: hive-operator/tests/kuttl/smoke/01-install-postgres.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+  - script: >-
+      helm install hive --version=10
```
Comment (sbernauer) [inline hive-operator/tests/kuttl/smoke/01-install-postgres.yaml:6]:
```suggestion
      helm install hive
      --version=10
```

File: hive-operator/tests/kuttl/smoke/10-install-hive.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+metadata:
+  name: test-hive-postgres
+spec:
+  version: 2.3.9
+  s3:
+    inline:
+      host: test-minio
+      port: 9000
+      accessStyle: Path
+      secretClass: test-hive-s3-secret
```
Comment (sbernauer) [inline hive-operator/tests/kuttl/smoke/10-install-hive.yaml:13]: Sorry to bring that up here, it's just the first time me noticing this:
We need to decide if we want a SecretClass from the secret-operator (more flexibility for e.g. automatically retrieve session token) or a plain k8s secret (easier).
Current the name is `secretClass` beside not beeing a `SecretClass`, but a secret

File: hive-operator/tests/kuttl/smoke/00-assert.yaml (hunk 1 above)
Comment (maltesander) [reply hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: No that is not how it works. The Service is only created after the pod is running i believe. There is no stateful set i can check for and the pod name has random characters attached.

File: hive-operator/tests/kuttl/smoke/10-install-hive.yaml (hunk 3 above)
Comment (maltesander) [reply hive-operator/tests/kuttl/smoke/10-install-hive.yaml:13]: That makes sense, but we currently do not have anything like that in the secret operator? I only see K8sSearch and AutoTls? This is just about providing access and secret key?

**Comment by sbernauer** [comment]
When i run the tests with `hive=0.6.0-pr179` i get the following expectation. Can you reproduce that?
```
    logger.go:42: 09:02:43 | smoke/30-prepare-test-metastore | running command: [sh -c kubectl exec -n $NAMESPACE test-metastore-0 -- python /tmp/test_metastore.py -n $NAMESPACE]
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore | Traceback (most recent call last):
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |   File "/tmp/test_metastore.py", line 71, in <module>
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |     hive_client.create_external_table(table(database_name, s3_test_table_name, "s3a://test/"))
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |   File "/root/.local/lib/python3.10/site-packages/hive_metastore_client/hive_metastore_client.py", line 218, in create_external_table
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |     self.create_table(table)
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |   File "/root/.local/lib/python3.10/site-packages/thrift_files/libraries/thrift_hive_metastore_client/ThriftHiveMetastore.py", line 2633, in create_table
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |     self.recv_create_table()
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |   File "/root/.local/lib/python3.10/site-packages/thrift_files/libraries/thrift_hive_metastore_client/ThriftHiveMetastore.py", line 2659, in recv_create_table
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore |     raise result.o3
    logger.go:42: 09:02:45 | smoke/30-prepare-test-metastore | thrift_files.libraries.thrift_hive_metastore_client.ttypes.MetaException: MetaException(message='Got exception: java.io.InterruptedIOException doesBucketExist on test: com.amazonaws.AmazonClientException: No AWS Credentials provided by BasicAWSCredentialsProvider EnvironmentVariableCredentialsProvider InstanceProfileCredentialsProvider : com.amazonaws.SdkClientException: Failed to connect to service endpoint: ')

File: hive-operator/tests/kuttl/smoke/00-assert.yaml (hunk 1 above)
Comment (sbernauer) [reply hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: I tested it out and i think we can remove 00-assert and 01-assert entirely. The `helm install` should wait for deployment to be ready

Comment (sbernauer) [reply hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: FYI We could also assert for the Deployment with something like
```
apiVersion: kuttl.dev/v1beta1
kind: TestAssert
commands:
- command: kubectl assert exist-enhanced deployment example-deployment -n $NAMESPACE --field-selector status.readyReplicas=3
```
But not very nice if you ask me

**Comment by sbernauer** [comment]
I wasn't able to reproduce the error. I think it was my fault, you can ignore my comment

File: hive-operator/tests/kuttl/smoke/00-assert.yaml (hunk 1 above)
Comment (sbernauer) [reply hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: Ok i tested a little bit at came up with the following solution:
* You can entirely remove `00-assert.yaml` as the helm install minio waits till minio is ready an all the needed buckets and users are created
* `01-assert.yaml` looks as follows as the helm install postgres does not wait but produces a regular sts
```
---
apiVersion: kuttl.dev/v1beta1
kind: TestAssert
metadata:
  name: install-postgres
timeout: 600
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: hive-postgresql
status:
  readyReplicas: 1
  replicas: 1
```

File: hive-operator/tests/kuttl/smoke/10-install-hive.yaml (hunk 3 above)
Comment (sbernauer) [reply hive-operator/tests/kuttl/smoke/10-install-hive.yaml:13]: We have a meeting discussing this today at 15:00

File: hive-operator/tests/kuttl/smoke/00-assert.yaml (hunk 1 above)
Comment (maltesander) [reply hive-operator/tests/kuttl/smoke/00-assert.yaml:8]: You definitely need the 01-assert. The metastore needs to run the schema tool which fails if the database is not reachable yet. I tested without the 00-assert and that works.

**Comment by sbernauer** [comment]
LGTM beside the switch to the secretClass :+1:

**Review by sbernauer — APPROVED** [review]
works for me

**Pull request merged**